doc/ImplementingApplications/ConceptOfElasticsearch/ConceptOfElasticsearch.md

```diff
@@ -0,0 +1,132 @@
+## Elasticsearch (ES)
+
+**Disclaimer**: This guide is not meant to be a manual for ES itself,
+it only covers areas implemented in ApplicationPattern and guides how to use
+the class ElasticsearchService in your application. For more information about
+Elasticsearch, go to [official documentation](https://www.elastic.co/guide/en/elasticsearch/reference/7.17/index.html).
+
+**Note**: The version of Elasticsearch used in ApplicationPattern is 7.17 and so all the documentation from official website should be
+looked at with this version in mind.
+
+### What is ES and why should my application use it?
+
+ES is a distributed, free and open search and analytics engine for all types of data, including textual, numerical, geospatial, structured, and unstructured. Your application should consider using it, if:
+- it will store large amounts of data
+- it will store data often
+
+### Brief introduction to ES buzzwords
+
+Basic building block in ES is `document`. By this word, ES understands one item of `data`. Elasticsearch stores complex data structures that have been serialized as JSON documents.
```

Comment:
can we edit buzzwords in the below format 
buzzword : description of buzzword

Reply:
Done.